### PR TITLE
[cdc-common] add createFieldGetters method to create FieldGetters of given Schema.

### PR DESCRIPTION
**Pull request by lvyanquan**
This pull request is a follow-up of https://github.com/ververica/flink-cdc-connectors/pull/2734. 
We now remove GenericRecordData and introduce new BinaryRecordData, so we can't get Field for RecordData directly and need to use FieldGetter.
This pr provide a util method to create FieldGetter and an example to get fields from RecordData.

### CODE REVIEW

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-values/src/main/java/com/ververica/cdc/connectors/values/sink/ValuesDataSinkHelper.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            DataChangeEvent dataChangeEvent = (DataChangeEvent) event;
+            String eventStr =
+                    "DataChangeEvent{"
+                            + "tableId="
+                            + dataChangeEvent.tableId()
+                            + ", before="
+                            + tableSchemaState.getFields(dataChangeEvent.before())
+                            + ", after="
+                            + tableSchemaState.getFields(dataChangeEvent.after())
+                            + ", op="
```
Comment (leonardBang) [inline flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-values/src/main/java/com/ververica/cdc/connectors/values/sink/ValuesDataSinkHelper.java:41]: Could you add a tests for this change ?

**Review by leonardBang — COMMENTED** [review]
Thanks @lvyanquan for the contribution, left one minor coments

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-values/src/main/java/com/ververica/cdc/connectors/values/sink/ValuesDataSinkHelper.java (hunk 1 above)
Comment (lvyanquan) [reply flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-values/src/main/java/com/ververica/cdc/connectors/values/sink/ValuesDataSinkHelper.java:41]: Address it and add in ValuesDataSinkHelperTest.

**Comment by leonardBang** [comment]
> the failed CI would be rerun after #2737 merged.

I've merged #2737

**Comment by lvyanquan** [comment]
The left of CI failure is unrelated with this pr.

**Comment by lvyanquan** [comment]
@leonardBang Reopen it to let ValuesSink not implement StatefulSink and remove class like TableSchemaState.
PTAL again.

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-values/src/main/java/com/ververica/cdc/connectors/values/sink/ValuesDataSink.java [hunk 2]
```diff
@@ -66,7 +62,7 @@ public MetadataApplier getMetadataApplier() {
     }
 
     /** an e2e {@link Sink} implementation that print all {@link DataChangeEvent} out. */
-    private static class ValuesSink implements StatefulSink<Event, TableSchemaState> {
```
Comment (leonardBang) [inline flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-values/src/main/java/com/ververica/cdc/connectors/values/sink/ValuesDataSink.java:69]: TableSchemaState could remove now ?

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-values/src/main/java/com/ververica/cdc/connectors/values/sink/ValuesDataSink.java [hunk 3]
```diff
@@ -109,48 +89,47 @@ private static class ValuesSinkWriter extends PrintSinkOutputWriter<Event>
...
                             SchemaUtils.applySchemaChangeEvent(
-                                    schemaMaps.get(tableId), schemaChangeEvent));
+                                    schemaMaps.get(tableId), schemaChangeEvent);
+                    schemaMaps.put(tableId, schema);
+                    fieldGetterMaps.put(tableId, SchemaUtils.createFieldGetters(schema));
                 }
             } else if (materializedInMemory && event instanceof DataChangeEvent) {
                 ValuesDatabase.applyDataChangeEvent((DataChangeEvent) event);
             }
+            // print the detail message to console for verification.
```
Comment (leonardBang) [inline flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-values/src/main/java/com/ververica/cdc/connectors/values/sink/ValuesDataSink.java:123]: Should we add a config to control print or not ?

Comment (lvyanquan) [reply flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-values/src/main/java/com/ververica/cdc/connectors/values/sink/ValuesDataSink.java]: address it.

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-values/src/main/java/com/ververica/cdc/connectors/values/sink/ValuesDataSink.java (hunk 2 above)
Comment (lvyanquan) [reply flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-values/src/main/java/com/ververica/cdc/connectors/values/sink/ValuesDataSink.java:69]: Yeah, we don't need to add Schema message to State as CreateTableEvent will be sent firstly.

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-values/src/main/java/com/ververica/cdc/connectors/values/sink/ValuesDataSinkOptions.java [hunk 4]
```diff
@@ -28,4 +28,10 @@ public class ValuesDataSinkOptions {
                     .defaultValue(false)
                     .withDescription(
                             "True if the DataChangeEvent need to be materialized in memory.");
+
+    public static final ConfigOption<Boolean> PRINT =
+            ConfigOptions.key("print")
```
Comment (leonardBang) [inline flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-values/src/main/java/com/ververica/cdc/connectors/values/sink/ValuesDataSinkOptions.java:33]: print.enable ?

Comment (lvyanquan) [reply flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-values/src/main/java/com/ververica/cdc/connectors/values/sink/ValuesDataSinkOptions.java]: done.

**Review by leonardBang — APPROVED** [review]
Thanks @lvyanquan for the update, LGTM

**Pull request merged**